File: FortLauncher/FortLauncher/Services/Utils/FortniteDetect.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FortLauncher.Services.Utils
{
    public class FortniteDetect
    {
        public static string Init(string BuildString)
        {
            if (BuildString.Contains("3724489"))
            {
                BuildString = "1.8";
            }
            else if (BuildString.Contains("3807424"))
            {
                BuildString = "1.11";
            }
            else if (BuildString.Contains("3870737"))
            {
                BuildString = "2.4.2";
            }
            else if (BuildString.Contains("3741772"))
            {
                BuildString = "1.8.2";
            }
            else if (BuildString.Contains("3240987"))
            {
                BuildString = "Alpha";
            }
            else
            {
                if (BuildString.Contains("-"))
                {
                    var BuildAdding = BuildString.Split("-");
                    if (BuildAdding.Length >= 1)
                    {
                        BuildString = BuildAdding[1];
                    }
                    else
                    {
                        BuildString = "Unknown";
                    }
                }
                else
                {
                    BuildString = "Unknown?";
                }
            }
            return BuildString;
        }
    }
}

File: FortLauncher/FortLauncherV2/backend/FortniteDetect.ts
export function FortniteDetect(buildString: string): string {
    if (buildString.includes('3724489')) {
      buildString = '1.8'
    }
    else if (buildString.includes('3700114')) {
      buildString = '1.7.2'
    } else if (buildString.includes('3807424')) {
      buildString = '1.11'
    } else if (buildString.includes('3870737')) {
      buildString = '2.4.2'
    } else if (buildString.includes('3741772')) {
      buildString = '1.8.2'
    } else if (buildString.includes('3240987')) {
      buildString = 'Alpha'
    } else {
      if (buildString.includes('-')) {
        const buildAdding = buildString.split('-')
  
        if (buildAdding.length >= 2) {
          buildString = buildAdding.slice(1).join('-')
        } else {
          buildString = 'Unknown'
        }
      } else {
        buildString = 'Unknown?'
      }
    }
    return buildString
  }